figures/scripts/transcriptomic-overview.R

```diff
@@ -0,0 +1,336 @@
+                                              "gradient_color_palette.tsv"))
+
+#### UMAP plot -----------------------------------------------------------------
+
+dim_red_dir <- file.path(
+  analyses_dir,
+  "transcriptomic-dimension-reduction"
+)
+
+source(file.path(dim_red_dir, "util", "dimension-reduction-functions.R"))
+
+rsem_umap_file <- file.path(
+  dim_red_dir,
+  "results",
+  "rsem_stranded_log_umap_scores_aligned.tsv"
+)
+
+umap_plot <- read_tsv(rsem_umap_file) %>%
+  select(X1, X2, short_histology) %>%
+  replace_na(list(short_histology = "none")) %>%
+  plot_dimension_reduction(point_color = "short_histology",
+                           x_label = "UMAP1",
+                           y_label = "UMAP2",
+                           color_palette = histology_palette) +
+  theme(text = element_text(size = 15))
+
+# Save temporary PNG
+ggsave(umap_png, plot = umap_plot, width = 4, height = 4)
+
+#### GSVA scores ---------------------------------------------------------------
+
+gsva_dir <- file.path(
+  analyses_dir,
+  "gene-set-enrichment-analysis"
+)
+
+gsva_scores_file <- file.path(
+  gsva_dir,
+  "results",
+  "gsva_scores_stranded.tsv"
+)
+
+gsva_stats_file <- file.path(
+  gsva_dir,
+  "results",
+  "gsva_anova_stranded_short_histology.tsv"
+)
+
+gsva_scores_df <- read_tsv(gsva_scores_file)
+gsva_stats_df <- read_tsv(gsva_stats_file)
+
+gsva_scores_mat <- gsva_scores_df %>%
+  spread(Kids_First_Biospecimen_ID, gsea_score) %>%
+  mutate(hallmark_name = stringr::str_replace_all(
+    stringr::str_remove(hallmark_name, "HALLMARK_"), "_", " ")
+  ) %>%
+  tibble::column_to_rownames("hallmark_name") %>%
+  as.matrix()
+
+# What gene sets are we going to include in our heatmap?
+# We'll filter based on whether the ANOVA has a significant p-value
+included_genesets <- gsva_stats_df %>%
+  mutate(hallmark_name = stringr::str_replace_all(
+    stringr::str_remove(hallmark_name, "HALLMARK_"), "_", " ")
+  ) %>%
+  filter(significant_anova) %>%
+  pull(hallmark_name)
+
+# Filter to only those gene sets
+gsva_scores_mat <- gsva_scores_mat[included_genesets, ]
+
+# Let's deal with the color palette for the heatmap we will make
+na_color <- divergent_palette %>%
+  filter(color_names == "na_color")
+
+divergent_palette  <- divergent_palette %>%
+  filter(color_names != "na_color")
+
+divergent_col_val <- seq(from = min(gsva_scores_mat),
+                         to = max(gsva_scores_mat),
+                         length.out = nrow(divergent_palette))
+
+gsva_col_fun <- circlize::colorRamp2(divergent_col_val,
+                                     divergent_palette$hex_codes)
+
+# Annotation bar for short histology
+# First get a data frame for the annotation and ensure samples are in the
+# same order
+short_histology_df <- histologies_df %>%
+  filter(experimental_strategy == "RNA-Seq",
+                RNA_library == "stranded") %>%
+  select(Kids_First_Biospecimen_ID,
+                short_histology) %>%
```

Comment:
I think to make the `HeatmapAnnotation` have the correct label name you need to set it as the column name here, so something like?
```suggestion
         "short histology" = short_histology) %>%
```

Reply:
Also, indentation here 😉